internal/resolution/entity_sources/catalogsource/catsrc_v1beta1_client.go

```diff
@@ -0,0 +1,285 @@
+package catalogsource
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sort"
+	"strings"
+	"sync"
+
+	catsrcV1 "github.com/operator-framework/catalogd/pkg/apis/core/v1beta1"
+	"github.com/operator-framework/deppy/pkg/deppy"
+	"github.com/operator-framework/deppy/pkg/deppy/input"
+	"github.com/operator-framework/operator-registry/alpha/property"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type v1beta1CatalogSourceConnector struct {
+	sync.RWMutex
+	client client.Client
+}
+
+func NewV1beta1CatalogSourceConnector(cl client.Client) *v1beta1CatalogSourceConnector {
+	return &v1beta1CatalogSourceConnector{
+		RWMutex: sync.RWMutex{},
+		client:  cl,
+	}
+}
+
+// id --> metadata.Name
+func (c *v1beta1CatalogSourceConnector) Get(ctx context.Context, id deppy.Identifier) *input.Entity {
```

Comment:
Is this part of a deppy interface that we needed to implement? If so, it seems like that interface needs to be updated to accomadate an error being returned.

Reply:
yes, this is a deppy interface. The `EntitySource` interface needs to be modified in here (https://github.com/operator-framework/deppy/blob/dc02e928470f892984255923c1cc449fe4a574e8/pkg/deppy/input/entity_source.go#L27) to return an error